src/VisualStudio/TestUtilities/IntegrationHelper.cs

```diff
@@ -0,0 +1,241 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+using EnvDTE;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.VisualStudio.OLE.Interop;
+using Microsoft.Win32;
+using Roslyn.VisualStudio.Test.Utilities.Interop;
+
+using Process = System.Diagnostics.Process;
+
+namespace Roslyn.VisualStudio.Test.Utilities
+{
+    internal static class IntegrationHelper
+    {
+        public static void CreateDirectory(string path, bool deleteExisting = false)
+        {
+            if (deleteExisting)
+            {
+                DeleteDirectoryRecursively(path);
+            }
+
+            Directory.CreateDirectory(path);
+        }
+
+        public static void DeleteDirectoryRecursively(string path)
+        {
+            if (Directory.Exists(path))
+            {
+                DirectoryExtensions.DeleteRecursively(path);
+            }
+        }
+
+        public static async Task DownloadFileAsync(string downloadUrl, string fileName)
+        {
+            using (var webClient = new WebClient())
+            {
+                await webClient.DownloadFileTaskAsync(downloadUrl, fileName).ConfigureAwait(continueOnCapturedContext: false);
+            }
+        }
+
+        public static IntPtr GetModalWindowFromParentWindow(IntPtr parentWindow)
```

Comment:
This is, locate a modal dialog that is blocking interaction with the specified window.